torch/utils/_debug_mode.py

```diff
@@ -204,7 +204,11 @@ def hash_tensor_fn(
     else:
         t_clean = t.to(dtype=torch.int64)
 
-    out = torch.hash_tensor(t_clean)
+    if t.numel() > 0:
+        out = torch.hash_tensor(t_clean)
+    else:
+        out = torch.tensor(0)
```

Comment:
you probably still want to avoid sync here, `out = torch.zeros((), device=t_clean.device)`